doc/source/theory/fluid_dynamics/linear_solvers.rst

```diff
@@ -1,4 +1,154 @@
+
+
+General theory on Krylov methods
+--------------------------------
+
+Suppose that :math:`x_0` is an initial guess for the solution of :math:`x`, and define the residual :math:`r_0=b - \mathcal{A} x_0`. Then, Krylov subspace methods are methods whose :math:`k`-th iterate :math:`x_k` satisfies:
+
+.. math::
+
+    x_k \in x_0 + \underbrace{\textrm{span} \{ x_0, \mathcal{A}x_0, \dotsc, \mathcal{A}^{k-1}x_0 \}}_{\mathcal{K}_k(\mathcal{A},x_0)}
+
+where :math:`\mathcal{K}_k(\mathcal{A},x_0)` is the :math:`k`-th Krylov subspace generated by :math:`\mathcal{A}` and :math:`x_0`. Because of the :math:`k` degrees of freedom in the choice of the iterate :math:`x_k`, :math:`k` constraints are required to make :math:`x_k` unique. This is achieved by imposing that the :math:`k`-th residual :math:`r_k = b - \mathcal{A} x_k` is orthogonal to a :math:`k`-dimensional space :math:`\mathcal{C}_k`, called the constraints space:
+
+.. math::
+
+    r_k = b - \mathcal{A} x_k \in r_0 + \mathcal{A} \mathcal{K}_k (\mathcal{A}, r_0), \hspace{2cm} r_k \perp \mathcal{C}_k
+
+By knowing the properties of :math:`\mathcal{A}`, it is possible to determine constraints spaces :math:`\mathcal{C}_k` that lead to uniquely defined iterates :math:`x_k` for :math:`k=1,2,\dotsc`. The choice of :math:`\mathcal{C}_k` and subsequent implementation leads to a different method.
+
+
+Generalized Minimal Residual Method (GMRES)
+_____________________________________________
+
+If :math:`\mathcal{A}` is non-symmetric positive definite, an approximate solution can be found by choosing :math:`\mathcal{C}_k = \mathcal{K}_k(\mathcal{A}, r_0)`, one implementation of this corresponds to the GMRES method. 
+
+
+*  The GMRES method is based on what is called an Arnoldi iteration. This iteration has a :math:`k`-term recurrence, which means that its computational cost increases as :math:`k` increases. The method is not very practical for very large :math:`k` because of the increase on memory and computational requirements. Therefore, for these kinds of systems, usually a restart is needed after certain iterations (for example 30 iterations) to build a new Krylov subspace. This restart may affect the convergence of the method when the matrix is not positive definite; a remedy for this is the use of a preconditioner.
```

Comment:
I know the default number of Krylov vectors of our GMRES vector is around 30. But in our case, using a low number of Krylov vectors like this leads to stagnation in a lot of applications.  By default, now our solver uses 200 or 250 instead of 30, if I remember correctly. Writing that here may suggest to the user that using 30 Krylov vectors would be ok in our solver when in a high number of applications, it is not ok.

Reply:
True, the default value in the parameters is actually 100 or 200